homeassistant/components/music_assistant/config_flow.py

```diff
@@ -0,0 +1,168 @@
+
+        This flow is triggered by the Zeroconf component. It will check if the
+        host is already configured and delegate to the import step if not.
+        """
+        # abort if discovery info is not what we expect
+        if "server_id" not in discovery_info.properties:
+            return self.async_abort(reason="missing_server_id")  # type: ignore[no-any-return]
+        # abort if we already have exactly this server_id
+        # reload the integration if the host got updated
+        server_id = discovery_info.properties["server_id"]
+        base_url = discovery_info.properties["base_url"]
+        await self.async_set_unique_id(server_id)
+        self._abort_if_unique_id_configured(
+            updates={CONF_URL: base_url},
+            reload_on_update=True,
+        )
+        self.server_info = ServerInfoMessage.from_dict(discovery_info.properties)
+        return await self.async_step_discovery_confirm()
+
+    async def async_step_discovery_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle user-confirmation of discovered server."""
+        if TYPE_CHECKING:
+            assert self.server_info is not None
+        if user_input is not None:
+            # Check that we can connect to the address.
+            try:
+                await get_server_info(self.hass, self.server_info.base_url)
+            except CannotConnect:
+                return self.async_abort(reason="cannot_connect")  # type: ignore[no-any-return]
+            return await self._async_create_entry_or_abort()
+        return self.async_show_form(  # type: ignore[no-any-return]
+            step_id="discovery_confirm",
+            description_placeholders={"url": self.server_info.base_url},
+        )
+
+    async def _async_create_entry_or_abort(self) -> ConfigFlowResult:
+        """Return a config entry for the flow or abort if already configured."""
+        assert self.server_info is not None
+
+        for config_entry in self._async_current_entries():
+            if config_entry.unique_id != self.server_info.server_id:
+                continue
+            self.hass.config_entries.async_update_entry(
+                config_entry,
+                data={
+                    **config_entry.data,
+                    CONF_URL: self.server_info.base_url,
+                },
+                title=DEFAULT_TITLE,
+            )
+            await self.hass.config_entries.async_reload(config_entry.entry_id)
+            raise AbortFlow("reconfiguration_successful")
+
+        # Abort any other flows that may be in progress
+        for progress in self._async_in_progress():
+            self.hass.config_entries.flow.async_abort(progress["flow_id"])
+
+        return self.async_create_entry(  # type: ignore[no-any-return]
+            title=DEFAULT_TITLE,
+            data={
+                CONF_URL: self.server_info.base_url,
+            },
+        )
```

Comment:
Looks like this is also doing reconfigure? Let's split this logic back into the user and zeroconf step